### PR TITLE
Remove Accumulable

### PR DESCRIPTION
**Pull request by jackgoldsmith4**

### CODE REVIEW

File: src/main/scala/is/hail/io/gen/GenLoader.scala [hunk 1]
```diff
@@ -124,11 +75,7 @@ object GenLoader {
       val d1 = gp(i + 1)
       val d2 = gp(i + 2)
       val sumDosages = d0 + d1 + d2
-      if (sumDosages == 0.0)
-        reportAcc += GenReport.gpNoCall
-      else if (math.abs(sumDosages - 1.0) > tolerance)
-        reportAcc += GenReport.gpSumLessThanTolerance
-      else {
+      if (sumDosages != 0.0 && !(math.abs(sumDosages - 1.0) > tolerance)) {
```
Comment (jigold) [inline src/main/scala/is/hail/io/gen/GenLoader.scala]: You can simplify this:
`if (math.abs(sumDosages - 1.0) <= tolerance)`

File: src/main/scala/is/hail/io/gen/GenLoader.scala [hunk 2]
```diff
@@ -70,12 +25,9 @@ object GenLoader {
 
     val nSamples = sampleIds.length
 
-    val reportAcc = sc.accumulable[mutable.Map[Int, Int], Int](mutable.Map.empty[Int, Int])
```
Comment (jigold) [inline src/main/scala/is/hail/io/gen/GenLoader.scala]: Double check that the accumulable is also gone from the `LoadBgen` class.

File: src/main/scala/is/hail/io/vcf/HtsjdkRecordReader.scala [hunk 3]
```diff
@@ -178,18 +174,15 @@ case class GenotypeRecordReader(vcfSettings: VCFSettings) extends HtsjdkRecordRe
 
       val ad = g.getAD
       if (g.hasAD) {
-        if (vcfSettings.skipBadAD && ad.length != nAlleles)
-          reportAcc += VCFReport.ADInvalidNumber
-        else
+        if (!vcfSettings.skipBadAD || ad.length == nAlleles)
```
Comment (jigold) [inline src/main/scala/is/hail/io/vcf/HtsjdkRecordReader.scala]: I think this is clearer:

`if (ad.length == nAlleles || !vcfSettings.skipBadAD)`

File: src/main/scala/is/hail/io/vcf/HtsjdkRecordReader.scala [hunk 4]
```diff
@@ -228,11 +217,9 @@ case class GenotypeRecordReader(vcfSettings: VCFSettings) extends HtsjdkRecordRe
           if (!g.hasDP)
             gb.setDP(adsum + od)
           else if (!filter && adsum + od != g.getDP) {
```
Comment (jigold) [inline src/main/scala/is/hail/io/vcf/HtsjdkRecordReader.scala]: I thin you can get rid of the checks for `filter` here.

File: src/main/scala/is/hail/io/vcf/HtsjdkRecordReader.scala [hunk 5]
```diff
@@ -228,11 +217,9 @@ case class GenotypeRecordReader(vcfSettings: VCFSettings) extends HtsjdkRecordRe
...
-            reportAcc += VCFReport.ADODDPPMismatch
             filter = true
           }
         } else if (!filter) {
```
Comment (jigold) [inline src/main/scala/is/hail/io/vcf/HtsjdkRecordReader.scala]: this can be just `else`

**Comment by cseed** [comment]
Looks like you missed some references to the *Report objects in HailContext.

**Comment by jackgoldsmith4** [comment]
@cseed ripping out the references to Report (DuplicateReport) objects screws up the Python interface because it supports the method deduplicate() in VariantSampleMatrix, which initializes a Report object

**Comment by tpoterba** [comment]
Deduplicate should still be there and work, just not report how many duplicates there were.

**Comment by tpoterba** [comment]
I should have the 0.1 / devel stuff set up by the end of the day, and we can make this PR against the development branch, where you can change whatever you want.

**Comment by jackgoldsmith4** [comment]
@jigold

File: src/main/scala/is/hail/io/gen/GenLoader.scala [hunk 6]
```diff
@@ -124,11 +75,7 @@ object GenLoader {
       val d1 = gp(i + 1)
       val d2 = gp(i + 2)
       val sumDosages = d0 + d1 + d2
-      if (sumDosages == 0.0)
-        reportAcc += GenReport.gpNoCall
-      else if (math.abs(sumDosages - 1.0) > tolerance)
-        reportAcc += GenReport.gpSumLessThanTolerance
-      else {
+      if (sumDosages != 0.0 && math.abs(sumDosages - 1.0) <= tolerance) {
```
Comment (jigold) [inline src/main/scala/is/hail/io/gen/GenLoader.scala]: You don't need the `sumDosages != 0` check. This will be caught by the tolerance test.

**Comment by cseed** [comment]
@tpoterba devel has landed in master and it like report functions have been removed.  Is this good to go?

**Comment by jackgoldsmith4** [comment]
@tpoterba I think I still have to remove the report method from Python

**Comment by tpoterba** [comment]
Ah, do that first. No reason not to, now.

**Pull request merged**